bzr.go

```diff
@@ -146,6 +147,40 @@ func (s *BzrRepo) Tags() ([]string, error) {
 	return tags, nil
 }
 
+// CurrentVersionsWithRevs returns a list of available branches and tags, and
+// their underlying revision identifiers. The list is guaranteed to be current.
+//
+// As the model we use for bzr interaction (and go get generally does as well)
+// involves interacting only with a bzr repository's default branch, we return
+// only tags here.
+func (s *BzrRepo) CurrentVersionsWithRevs() (versions []VersionInfo, localSynced bool, err error) {
+	var out []byte
+	// Update the local first
+	_, err = s.runFromDir("bzr", "pull")
+	if err != nil {
+		return
+	}
+	localSynced = true
```

Comment:
Yeah, this is one of those cases where trying to bridge across the different vcs models gets a little nasty. The issue is really just git: while we *could* encode update failure as an error, and that would be sufficient and work fine for bzr and hg, it wouldn't for git - when `git ls-remote` succeeds, it won't update the local repo, but that's still a non-erroring return.